docs/source/python/dataset.rst

```diff
@@ -0,0 +1,354 @@
+
+.. ipython:: python
+
+    dataset.to_table(filter=ds.field("part") == "b").to_pandas()
+
+
+Different partitioning schemes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The above example uses a hive-like directory scheme, such as "/year=2009/month=11/day=15".
+We specified this passing the ``partitioning="hive"`` keyword. In this case,
+the types of the partition keys are inferred from the file paths.
+
+It is also possible to explicitly define the schema of the partition keys
+using the :func:`partitioning` function. For example:
+
+.. code-block::
+
+    part = ds.partitioning(
+        pa.schema([("year", pa.int16()), ("month", pa.int8()), ("day", pa.int32())]),
+        flavor="hive"
+    )
+    dataset = ds.dataset(..., partitioning=part)
+
+In addition to the hive-like directory scheme, also a "directory partitioning"
+scheme is supported, where the segments in the file path are the values of
+the partition keys without including the name. The equivalent (year, month, day)
+example would be "/2019/11/15/".
+
+Since the names are not included in the file paths, those need to be specified:
+
+.. code-block::
+
+    part = ds.partitioning(field_names=["year", "month", "day"])
+
+Also here, a full schema can be provided to not let the types be inferred
+from the file paths.
+
+
+Reading from cloud storage
+--------------------------
+
+- example with s3 filesystem / hdfs filesystem
+
+
+
+Manual specification of the Dataset
+-----------------------------------
+
+The :func:`dataset` function allows to easily create a Dataset from a directory,
+crawling all subdirectories for files and partitioning information. However,
+when the dataset files and partitions are already known (for example, when this
+information is stored in metadata), it is also possible to create a Dataset
+explicitly using the lower level API without any automatic discovery or
+inference.
+
+For the example here, we are going to use a dataset where the file names contain
+additional partitioning information:
+
+.. ipython:: python
+
+    # creating a dummy dataset: directory with two files
+    table = pa.table({'col1': range(3), 'col2': np.random.randn(3)})
+    pq.write_table(table, "parquet_dataset_manual/data_file1.parquet")
+    pq.write_table(table, "parquet_dataset_manual/data_file2.parquet")
+
+To create a Dataset from a list of files, we need to specify the schema, format,
+filesystem, and paths manually:
+
+.. ipython:: python
+
+    import pyarrow.fs
+
+    schema = pa.schema([("file", pa.int64()), ("col1", pa.int64()), ("col2", pa.float64())])
+
+    dataset = ds.FileSystemDataset(
+        schema, None, ds.ParquetFileFormat(), pa.fs.LocalFileSystem(),
+        ["parquet_dataset_manual/data_file1.parquet", "parquet_dataset_manual/data_file2.parquet"],
+        [ds.field('file') == 1, ds.field('file') == 2])
+
+We also specified the "partition expressions" for our files, so this information
+is included when reading the data and can be used for filtering:
```

Comment:
```suggestion
Since we specified the "partition expressions" for our files, this information
is materialized as columns when reading the data and can be used for filtering:
```